pkg/kubelet/network/cni/cni.go

```diff
@@ -43,6 +44,8 @@ type cniNetworkPlugin struct {
 	host           network.Host
 	execer         utilexec.Interface
 	nsenterPath    string
+	mu             sync.Mutex
+	podMutexes     map[string]*sync.Mutex
```

Comment:
So this PR should not be needed at all?  Can we guarantee that network
plugin calls are never in parallel?  If so, maybe the fix, @dcbw, is just
to document that :)

On Thu, Jul 28, 2016 at 5:27 PM, Yu-Ju Hong <notifications@github.com>
wrote:

> In pkg/kubelet/network/cni/cni.go
> <https://github.com/kubernetes/kubernetes/pull/29217#discussion_r72726419>
> :
>
> > @@ -43,6 +44,8 @@ type cniNetworkPlugin struct {
> >  	host           network.Host
> >  	execer         utilexec.Interface
> >  	nsenterPath    string
> > +	mu             sync.Mutex
> > +	podMutexes     map[string]*sync.Mutex
>
> For the normal, graceful deletions, thel pod worker (goroutine) is
> responsible for killing the pod, and there should not be race conditions.
> What @dcbw <https://github.com/dcbw> talked about is the case where a pod
> gets deleted non-gracefully (i.e., ---grace-period=0), and kubelet's
> global cleanup goroutine will try to kill the pod regardless of what the
> pod worker is doing. This race condition is the root cause and should be
> fixed: #13311 (comment)
> <https://github.com/kubernetes/kubernetes/issues/13311#issuecomment-229214387>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/29217/files/7125b3b8e2bc820896f2a25377b6402545208a4f#r72726419>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFVgVN15yi0c-vmh-YT316KX1Exb8gHZks5qaUjlgaJpZM4JP-o4>
> .
>